pkg/specgen/generate/kube/kube.go

```diff
@@ -53,6 +53,10 @@ func ToPodOpt(ctx context.Context, podName string, p entities.PodCreateOptions,
 	if podYAML.Spec.ShareProcessNamespace != nil && *podYAML.Spec.ShareProcessNamespace {
 		p.Share = append(p.Share, "pid")
 	}
+	// TODO: add check to make sure both share pid and hostpid are not specified
```

Comment:
@vrothberg do you think this check makes sense? and another log about telling the user that [it's an infra container but shares host pid](https://github.com/containers/podman/pull/17168/files#diff-cfef84d8077b70a186e602882259d5d3c6f8ecbf808b845e1f5a85c221df61e5R69)

Reply:
I do not feel strongly for or against adding a log.  If it would be helpful to you for debugging, it may be worth adding a debug log.

Reply:
Ok, I think I'll leave it at that then, it's a user-facing configuration option and makes sense for the user to decide how to use these.